compiler/noirc_frontend/src/lexer/errors.rs

```diff
@@ -64,7 +64,7 @@ impl LexerErrorKind {
 
                 (
                     "An unexpected character was found".to_string(),
-                    format!("Expected {expected}, but found {found}"),
+                    format!("Lexer: expected {expected}, but found {found}"),
```

Comment:
This reads a bit oddly to me to give lexer/parser details to users. Does a user care if a token was expected while lexing versus during parsing?

Reply:
Less that they care whether it's lexer/parser and more they care how easy it is to report. I've seen "lexing error" vs "parsing error" in certain languages, but if we could even add a `"."` at the end of the line the two errors would be distinguishable. Similarly, the lexing error could mention the word 'token' instead of saying 'Lexer.'

Reply:
The two errors should already be distinguishable because they are several lines, the first line of this error is "An unexpected character was found" which isn't present in the parser version.

The parser version is also slightly (though perhaps unintentionally) different. It says "Expected a foo but found bar" or "Expected an ack but found bar" where the lexer version doesn't have the a/an.

Reply:
I only bring this up because I think the words "lexer" and "parser" are jargon that users may not know. Since they aren't entirely relevant to the error, I'd like to avoid them to keep things simpler for those users.